Mtl igpu pmax

mkopec opened the pull request.

mkopec committed:
mb/clevo/mtl-h/ramstage.c: Set cnvi_wifi_core according to RF option

Fixes the following error message:
"CNVi WiFi is enabled without CNVi being enabled"

Upstream-Status: Inappropriate (baseboard still in review)

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h: Add variant-specific power limits

iGPU variant has a lower Psys PMAX value of 99W per schematic. Program
it correctly, to ensure power reporting scale is correct and Psys power
limits work as expected.

Upstream-Status: Inappropriate (board still in review)

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

miczyg1 commented on src/mainboard/clevo/mtl-h/variants/igpu/overridetree.cb:
```diff
@@ -1,4 +1,16 @@
+	register "power_limits_config[MTL_P_682_482_CORE]" = "{
+		.tdp_pl1_override = 35,
+		.tdp_pl2_override = 64,
+		.tdp_pl4 = 120,
```
Not sure if pl4 can be bigger than pmax. I know EC will limit it, but I don't know what the consequences may be.

How much Watts does the power supply have?

mkopec replied:
PSU is 90W, but Pmax has to be 99W because that's the scale that the battery charger uses to report Psys to CPU.

I thought PL4 can be up to PSU rating + max battery rating but it feels like every doc I read about it says something different. Will adjust PL4 to the same value as PMAX to be on the safe side.

mkopec replied:
https://github.com/Dasharo/coreboot/commit/ed3fdaec053cf201ed16a54a564b7455416b9bdf#diff-5440c50c876ba495f3ff1099c44d020892a1941250b8987c8aecea3d3e47d80fR7

miczyg1 replied:
@mkopec as I suspected. PsysPmax = PL4 + ROP MAX. See doc 634413 on our cloud (section 3.4). Basically we need a margin for Psys PMAX equivalent to Rest of Platform (ROP) power here.

> that's the scale that the battery charger uses to report Psys to CPU.

Scale or granularity? Where can I check it?

mkopec replied:
Scale, as determined by Rpsys resistor. See schematics page B-40 `MP29001 Controller`

So I guess we should reduce PL4 by ROP max, need to determine it somehow first

miczyg1 replied:
Fair enough. But if to believe what the doc says, PL4 should be lower than Psys PMAX. To be on the safe side, I would give PL4=90W maybe?

mkopec replied:
Done https://github.com/Dasharo/coreboot/pull/612/commits/86233ac56e4c18e398efd9b505924e29aeb9d029

mkopec committed:
mb/clevo/mtl-h/devicetree.cb: Remove smartamp device

The device is only present on the dGPU models, and is already enabled in
the dGPU overridetree. Remove from the baseboard devicetre, to fix
yellow bang in Windows 11 device manager.

Upstream-Status: Inappropriate (board still in review)

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec commented:
Fixes https://github.com/Dasharo/dasharo-issues/issues/1216

miczyg1 commented:
@mkopec needs rebase to fast-forward merge it

mkopec commented:
rebased, let's see what CI thinks
